TrackletAlgorithm/TrackMerger.cc

```diff
@@ -0,0 +1,176 @@
+
+  return;      
+}  
+
+
+void unloadTrack(
+  TrackStruct& aTrack,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::TrackWord> &trackWord_o,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_0_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_1_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_2_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_3_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_0_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_1_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_2_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_3_o 
+) {
+  #pragma HLS inline
+  trackWord_o.write(aTrack._trackWord);
+  barrelStubWords_0_o.write(aTrack._barrelStubArray[0][0]);
+  barrelStubWords_1_o.write(aTrack._barrelStubArray[1][0]);
+  barrelStubWords_2_o.write(aTrack._barrelStubArray[2][0]);
+  barrelStubWords_3_o.write(aTrack._barrelStubArray[3][0]);
+  diskStubWords_0_o.write(aTrack._diskStubArray[0][0]);
+  diskStubWords_1_o.write(aTrack._diskStubArray[1][0]);
+  diskStubWords_2_o.write(aTrack._diskStubArray[2][0]);
+  diskStubWords_3_o.write(aTrack._diskStubArray[3][0]);
+
+  return;
+}
+
+void TrackMerger(const BXType bx,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::TrackWord> &trackWord,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_0,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_1,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_2,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_3,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_0,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_1,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_2,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_3,
+  BXType bx_o,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::TrackWord> &trackWord_o,
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_0_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_1_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_2_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::BarrelStubWord> &barrelStubWords_3_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_0_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_1_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_2_o, 
+  hls::stream<TrackFit<NBarrelStub, NDiskStub>::DiskStubWord> &diskStubWords_3_o 
+  )
+{
+  #pragma HLS inline
```

Comment:
Would it make sense to use the `recursive` option here? It looks like you're inlining a lot of your subfunctions, and this would allow you to just have one instance of this pragma.

If the `recursive` option makes sense, maybe you could even move it to `TrackMergerTop` in TrackMergerTop.cc.